nucypher/blockchain/eth/interfaces.py

```diff
@@ -291,7 +291,7 @@ def attach_middleware(self):
         self.log.debug(f"Currently, our gas strategy returns a gas price of {gwei_gas_price} gwei")
 
         self.client.add_middleware(middleware.time_based_cache_middleware)
-        self.client.add_middleware(middleware.latest_block_based_cache_middleware)
+        # self.client.add_middleware(middleware.latest_block_based_cache_middleware)
```

Comment:
Remind me of the issue this line was causing? Did it cause some issues with testing?

Reply:
Issue with nonce, error occurred in upgradeability test

Reply:
Followed up in #2384